### PR TITLE
Kickoff

### PR DESCRIPTION
**Pull request by truccomarcos**
# Summary
- Kickoff
- Controllers
- Views
- Model
- MIgrations

# Trello Cards
- https://trello.com/c/aRAtMKsn/1-kick-off
- https://trello.com/c/Eyg5zCMv/3-mvc-controllers
- https://trello.com/c/g6yqSga6/4-mvc-views
- https://trello.com/c/DWUmdK9n/5-mvc-models
- https://trello.com/c/mcVxer9f/6-migrations

### CODE REVIEW

File: MvcMovie/.vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
+{
```
Comment (gussiciliano) [inline MvcMovie/.vscode/launch.json]: ignore all .vscode/ files

File: MvcMovie/Controllers/HelloWorldController.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+        public IActionResult Welcome(string name, int numTimes = 1)
+        {
+            ViewData["Message"] = "Hello " + name;
+            ViewData["NumTimes"] = numTimes;
+
+            return View();
+        }
+    }
+}
```
Comment (gussiciliano) [inline MvcMovie/Controllers/HelloWorldController.cs]: add empty line

File: MvcMovie/Controllers/HelloWorldController.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        }
+
+        // 
+        // GET: /HelloWorld/Welcome/ 
+
+        public IActionResult Welcome(string name, int numTimes = 1)
+        {
+            ViewData["Message"] = "Hello " + name;
+            ViewData["NumTimes"] = numTimes;
+
```
Comment (gussiciliano) [inline MvcMovie/Controllers/HelloWorldController.cs]: delete empty space

File: MvcMovie/Controllers/HelloWorldController.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    {
+        // 
+        // GET: /HelloWorld/
+
+        public IActionResult Index()
+        {
+            return View();
+        }
+
+        //
```
Comment (gussiciliano) [inline MvcMovie/Controllers/HelloWorldController.cs]: delete empty comment

File: MvcMovie/Controllers/HelloWorldController.cs [hunk 5]
```diff
@@ -0,0 +1,27 @@
+using Microsoft.AspNetCore.Mvc;
+using System.Text.Encodings.Web;
+
+namespace MvcMovie.Controllers
+{
+    public class HelloWorldController : Controller
+    {
+        //
```
Comment (gussiciliano) [inline MvcMovie/Controllers/HelloWorldController.cs]: delete empty comment

File: MvcMovie/Controllers/HelloWorldController.cs [hunk 6]
```diff
@@ -0,0 +1,27 @@
+using Microsoft.AspNetCore.Mvc;
+using System.Text.Encodings.Web;
+
+namespace MvcMovie.Controllers
+{
+    public class HelloWorldController : Controller
+    {
+        // 
+        // GET: /HelloWorld/
+
```
Comment (gussiciliano) [inline MvcMovie/Controllers/HelloWorldController.cs]: delete empty space

File: MvcMovie/Controllers/HelloWorldController.cs [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+        public IActionResult Index()
+        {
+            return View();
+        }
+
+        // 
+        // GET: /HelloWorld/Welcome/ 
+
```
Comment (gussiciliano) [inline MvcMovie/Controllers/HelloWorldController.cs]: delete empty space

File: MvcMovie/Controllers/HomeController.cs [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+    {
+        public IActionResult Index()
+        {
+            return View();
+        }
+
+        public IActionResult About()
+        {
+            ViewData["Message"] = "Your application description page.";
+
```
Comment (gussiciliano) [inline MvcMovie/Controllers/HomeController.cs]: delete empty space

File: MvcMovie/Controllers/HomeController.cs [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+            return View();
+        }
+
+        public IActionResult Contact()
+        {
+            ViewData["Message"] = "Your contact page.";
+
```
Comment (gussiciliano) [inline MvcMovie/Controllers/HomeController.cs]: delete empty space

File: MvcMovie/Models/Movie.cs [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+        [RegularExpression(@"^[A-Z]+[a-zA-Z''-'\s]*$"), Required, StringLength(30)]
+        public string Genre { get; set; }
+
+        [Range(1, 100), DataType(DataType.Currency)]
+        public decimal Price { get; set; }
+
+        [RegularExpression(@"^[A-Z]+[a-zA-Z''-'\s]*$"), StringLength(5)]
+        public string Rating { get; set; }
+    }
+}
```
Comment (gussiciliano) [inline MvcMovie/Models/Movie.cs]: add final line

File: MvcMovie/MvcMovie.csproj [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+    <PackageReference Include="Npgsql.EntityFrameworkCore.PostgreSQL" Version="1.1.0" />
+    <PackageReference Include="Npgsql.EntityFrameworkCore.PostgreSQL.Design" Version="1.1.0" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore.Design" Version="1.1.1" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore.Tools" Version="1.1.0" />
+
+
+  </ItemGroup>
+
+  <ItemGroup>
+    <DotNetCliToolReference Include="Microsoft.VisualStudio.Web.CodeGeneration.Tools" Version="1.0.1" />
```
Comment (gussiciliano) [inline MvcMovie/MvcMovie.csproj]: merge ItemGoup tags

File: MvcMovie/Program.cs [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+    public class Program
+    {
+        public static void Main(string[] args)
+        {
+            var host = new WebHostBuilder()
+                .UseKestrel()
+                .UseContentRoot(Directory.GetCurrentDirectory())
+                .UseIISIntegration()
+                .UseStartup<Startup>()
+
```
Comment (gussiciliano) [inline MvcMovie/Program.cs]: delete empty space

File: MvcMovie/Program.cs [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+        public static void Main(string[] args)
+        {
+            var host = new WebHostBuilder()
+                .UseKestrel()
+                .UseContentRoot(Directory.GetCurrentDirectory())
+                .UseIISIntegration()
+                .UseStartup<Startup>()
+                
+                .Build();
+
```
Comment (gussiciliano) [inline MvcMovie/Program.cs]: delete empty space

File: MvcMovie/Views/HelloWorld/Index.cshtml [hunk 14]
```diff
@@ -0,0 +1,7 @@
+@{
+    ViewData["Title"] = "Movie List";
+}
+
+<h2>My Movie List</h2>
+
+<p>Hello from our View Template!</p>
```
Comment (gussiciliano) [inline MvcMovie/Views/HelloWorld/Index.cshtml]: add final line

File: MvcMovie/Views/HelloWorld/Welcome.cshtml [hunk 15]
```diff
@@ -0,0 +1,12 @@
+@{
+    ViewData["Title"] = "Welcome";
+}
+
+<h2>Welcome</h2>
+
+<ul>
+    @for (int i = 0; i < (int)ViewData["NumTimes"]; i++)
```
Comment (gussiciliano) [inline MvcMovie/Views/HelloWorld/Welcome.cshtml:8]: why do you do this?
i don't remember the card jeje

File: MvcMovie/appsettings.json [hunk 16]
```diff
@@ -0,0 +1,11 @@
+﻿{
+  "Logging": {
+    "IncludeScopes": false,
+    "LogLevel": {
+      "Default": "Warning"
+    }
+  },
+  "DbContextSettings" :{
+    "DbConnectionString" : "User ID=postgres;Password=postgres;Host=localhost;Port=5432;Database=test_asp;Pooling=true;Timeout=1024;CommandTimeout=1024;"
+  }
```
Comment (gussiciliano) [inline MvcMovie/appsettings.json]: add final line

File: MvcMovie/bin/Debug/netcoreapp1.1/MvcMovie.runtimeconfig.dev.json [hunk 17]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (gussiciliano) [inline MvcMovie/bin/Debug/netcoreapp1.1/MvcMovie.runtimeconfig.dev.json]: ignore all bin files

File: MvcMovie/obj/Debug/netcoreapp1.1/CoreCompileInputs.cache [hunk 18]
```diff
@@ -0,0 +1 @@
+d7671eeb299a1b0216fa93fc9f90c61a58492715
```
Comment (gussiciliano) [inline MvcMovie/obj/Debug/netcoreapp1.1/CoreCompileInputs.cache]: ignore all obj files

**Review by gussiciliano — CHANGES_REQUESTED** [review]

File: MvcMovie/Views/HelloWorld/Welcome.cshtml (hunk 15 above)
Comment (truccomarcos) [reply MvcMovie/Views/HelloWorld/Welcome.cshtml:8]: I think it came from one of the tutorials

File: MvcMovie/.gitignore [hunk 19]
```diff
@@ -0,0 +1,3 @@
+bin/
+obj/
+.vscode/
```
Comment (gussiciliano) [inline MvcMovie/.gitignore]: add final line

File: MvcMovie/Views/HelloWorld/Welcome.cshtml (hunk 15 above)
Comment (gussiciliano) [reply MvcMovie/Views/HelloWorld/Welcome.cshtml:8]: oks :P

**Review by gussiciliano — APPROVED** [review]
just little comment :P

**Pull request merged**